weldx/schemas/weldx.bam.de/weldx/core/geometry/spatial_data-0.1.1.yaml

```diff
@@ -0,0 +1,90 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "asdf://weldx.bam.de/weldx/schemas/core/geometry/spatial_data-0.1.1"
+
+title: |
+  Schema for a point cloud.
+description: |
+  This schema describes a collection of data points in 3d space.
+
+examples:
+  -
+    - A simple `SpatialData` with triangulation
+    - |
+      !<asdf://weldx.bam.de/weldx/tags/core/geometry/spatial_data-0.1.1>
+        coordinates: !core/ndarray-1.0.0
+          data:
+          - [0.0, 0.0, 0.0]
+          - [1.0, 0.0, 0.0]
+          - [1.0, 1.0, 0.0]
+          - [0.0, 1.0, 0.0]
+          datatype: float64
+          shape: [4, 3]
+        triangles: !core/ndarray-1.0.0
+          data:
+          - [0, 1, 2]
+          - [2, 3, 0]
+          datatype: uint32
+          shape: [2, 3]
+  -
+    - A `SpatialData` object with data in three dimensions
+    - |
+      !<asdf://weldx.bam.de/weldx/tags/core/geometry/spatial_data-0.1.1>
+        coordinates: !<asdf://weldx.bam.de/weldx/tags/core/data_array-0.1.0>
+          attributes: {}
+          coordinates:
+          - !<asdf://weldx.bam.de/weldx/tags/core/variable-0.1.0>
+            name: c
+            dimensions: [c]
+            dtype: <U1
+            data: !core/ndarray-1.0.0
+              data: [x, y, z]
+              datatype: [ucs4, 1]
+              shape: [3]
+          data: !<asdf://weldx.bam.de/weldx/tags/core/variable-0.1.0>
+            name: data
+            dimensions: [p, n, c]
+            dtype: <f8
+            data: !core/ndarray-1.0.0
+              data:
+              - - [0.0, 0.0, 0.0]
+                - [0.0, 1.0, 0.0]
+              - - [1.0, 0.0, 1.0]
+                - [0.0, 1.0, 1.0]
+              datatype: float64
+              shape: [2, 2, 3]
+
+type: object
+properties:
+  coordinates:
+    description: |
+      The coordinates of the data points.
+    oneOf:
+      - tag: "tag:stsci.edu:asdf/core/ndarray-1.*"
+        properties:
+          datatype:
+            type: string
+            enum: [float32, float64]
+        wx_shape: [(~), 3]
+      - tag: "asdf://weldx.bam.de/weldx/tags/core/data_array-0.1.*"
+        wx_shape: [..., 3]
```

Comment:
agreed, in general I think it would be a good addition to have a powerful validator for xarray structures

the overall structure and implementation probably needs some planning

Reply:
Should I create a new issue as reminder?

Reply:
yes that never hurts :wink: